### PR TITLE
Update template and checks

### PR DESCRIPTION
**Pull request by jmesnil**
* Update the template to move the tracking issue and the stability level to the front matter
* When a PR is open, check that these fields are valid
* Generate a HTML page for the proposal with these fields.

This PR updates the template to be more descriptive about its expected content.

As a demo, I opened a "dummy" proposal in my fork that was missing the `issue` and `stability-level` in the front matter:
* https://github.com/jmesnil/wildfly-proposals/pull/14
  * The `missing-issue` and `invalid-stability-level` labels were automatically added to the PR
* I fixed my PR
  * The labels were removed as the checks passed
  * the asciidoc displays the front matter: https://github.com/jmesnil/wildfly-proposals/blob/main/bootable-jar/dummy_featyre.adoc
* The PR was merged and the metadata were added to the generated proposal page:
  * https://jmesnil.github.io/wildfly-proposals/bootable-jar/dummy_featyre.html 
* If the `stability-level` field is valid in the front matter, add a label to the GitHub PR of the form `stability-level/XXX` (as seen on my fork at https://github.com/jmesnil/wildfly-proposals/pull/15)

**Before that PR is merged, the following labels must be created:**
* `missing-issue` - The analysis document is missing an issue tracker in its metadata
* `invalid-stability-level` - The analysis document is missing a valid stability level in its metadata
* `stability-level/experimental` - Feature is at the "Experimental" stability level
* `stability-level/preview` - Feature is at the "Preview" stability level
* `stability-level/community` - Feature is at the "Community" stability level
* `stability-level/default` - Feature is at the "Default" stability level

### CODE REVIEW

File: design-doc-template.adoc [hunk 1]
```diff
@@ -4,125 +4,95 @@ categories:
 # - core
 # - management
 # if missing, add it to _data/widfly-categories and use its id
+#
+# Specify the stabitily level of the feature.
```
Comment (bstansberry) [inline design-doc-template.adoc:8]: s/stabitily/stability

File: design-doc-template.adoc [hunk 2]
```diff
@@ -4,125 +4,95 @@ categories:
...
+# Values can be one of: experimental preview community default
+stability-level: 
+# Specify the main issue of the feature
+# Most of the image, it will be a issue in https://issues.redhat.com/browse/WFLY
```
Comment (bstansberry) [inline design-doc-template.adoc:12]: s/image/time

File: design-doc-template.adoc [hunk 3]
```diff
@@ -4,125 +4,95 @@ categories:
...
 
 === Relevant Installation Types
-// Remove the x next to the relevant field if the feature in question is not relevant
-// to that kind of WildFly installation
-* [x] Traditional standalone server (unzipped or provisioned by Galleon)
-
-* [x] Managed domain
 
-* [x] OpenShift s2i
+__<List the installation  types thar are relevant for the features and remove any that is not relevant>__.
```
Comment (bstansberry) [inline design-doc-template.adoc:39]: s/is/are

File: design-doc-template.adoc [hunk 4]
```diff
@@ -4,125 +4,95 @@ categories:
...
 == Backwards Compatibility
 
-// Does this enhancement affect backwards compatibility with previously released
-// versions of WildFly?
-// Can the identified incompatibility be avoided?
+__<Does this enhancement affect backwards compatibility with previously released versions of WildFly? Can the identified incompatibility be avoided?>__
 
 === Default Configuration
 
+__<What is the impact of this feature in the default configuration(s) provided by WildFly?>__
```
Comment (bstansberry) [inline design-doc-template.adoc:64]: Note that this and the following items are specific areas to consider in terms incompatibility. They are not top level items for general description.

I'd like to have a discussion somewhere about the exact meaning of this one and the "Importing Existing Configuration" one as there are relationships. And they both need to be more fully described. For example we provide *standard* OOTB configurations, while configuration items, used or not in those standard configs, have default values. This one should primarily be about the latter, but there are various cases to consider.

File: design-doc-template.adoc [hunk 5]
```diff
@@ -4,125 +4,95 @@ categories:
...
+
 === Importing Existing Configuration
 
+TBD
+
 === Deployments
 
+__<What is the impact of this feature for deployments?>__
```
Comment (bstansberry) [inline design-doc-template.adoc:72]: "Does this feature change the behavior of deployments in incompatible ways?"

File: design-doc-template.adoc [hunk 6]
```diff
@@ -4,125 +4,95 @@ categories:
...
-== Security Considerations
+== Implementation Plan
 
-////
-Identification if any security implications that may need to be considered with this feature
-or a confirmation that there are no security implications to consider.
-////
+__<This section is optional. If you have a complex feature which can not be delivered all in one go, suggest the strategy.>__
+
+== Thread Model
```
Comment (bstansberry) [inline design-doc-template.adoc:82]: s/Thread/Threat

I'm not sure this is a better title than "Security Considerations". I imagine many devs only have a vague idea what thread modeling is.

File: design-doc-template.adoc [hunk 7]
```diff
@@ -4,125 +4,95 @@ categories:
...
-Release Note blog article for the release that first includes this feature. 
-Example article: http://wildfly.org/news/2018/08/30/WildFly14-Final-Released/.
-This content will be edited, so there is no need to make it perfect or discuss
-what release it appears in.  "See Overview" is acceptable if the overview is
-suitable. For simple features best covered as an item in a bullet-point list 
-of features containing a few words on each, use "Bullet point: <The few words>" 
-////
+
+__<Draft verbiage for up to a few sentences on the feature for inclusion in the Release Note blog article for the release that first includes this feature.__
+__Example article: http://wildfly.org/news/2018/08/30/WildFly14-Final-Released/.__
```
Comment (bstansberry) [inline design-doc-template.adoc:97]: Please change the link to https://www.wildfly.org/news/2024/01/25/WildFly31-Released/

That's a much more representative example of the way typical features are introduced.

File: design-doc-template.adoc [hunk 8]
```diff
@@ -4,125 +4,95 @@ categories:
...
+__This content will be edited, so there is no need to make it perfect or discuss what release it appears in. For simple features best covered as an item in a bullet-point list of features containing a few words on each, use "Bullet point: <The few words>">__
```
Comment (bstansberry) [inline design-doc-template.adoc:98]: I think that last sentence should be removed. I'm pretty sure I wrote it but I can't think of any features where that treatment was appropriate.

Also, thanks for dropping the 'See Overview'  bit. That option in practice is never more helpful than saying nothing here. The overviews are and should be written in a different way.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: design-doc-template.adoc (hunk 8 above)
Comment (jmesnil) [reply design-doc-template.adoc]: done

File: design-doc-template.adoc (hunk 4 above)
Comment (bstansberry) [reply design-doc-template.adoc:64]: I think this is what this should say:

__<Does this feature change the default value of any current configuration attributes? Does it change the configuration generated by any current Galleon layers?>__

File: design-doc-template.adoc [hunk 9]
```diff
@@ -4,125 +4,95 @@ categories:
...
-// Can the identified incompatibility be avoided?
+__<Does this enhancement affect backwards compatibility with previously released versions of WildFly? Can the identified incompatibility be avoided?>__
 
 === Default Configuration
 
+__<What is the impact of this feature in the default configuration(s) provided by WildFly?>__
+
 === Importing Existing Configuration
 
+TBD
```
Comment (bstansberry) [inline design-doc-template.adoc:68]: __<Does the proposed work affect the ability to run WildFly running an existing configuration? Is there anything else about the proposed work that may require changes to the WildFly server migration tool?>__

File: _includes/index-categories.html [hunk 10]
```diff
@@ -28,7 +28,13 @@ <h2 id="{{ cat.id }}">{{ cat.name }} <small>[{{ cat.id }}]</small></a></h2>
     {% for page in site.pages %}
     <ul>
       {% if page.categories contains cat.id %}
-      <li><a href="{{ site.baseurl }}{{ page.url }}">{{ page.title }}</a></li>
+      <li><a href="{{ site.baseurl }}{{ page.url }}">
+        {% if page.stability-level != "default" or page.stability-level != "community" %}
```
Comment (jmesnil) [inline _includes/index-categories.html:32]: @bstansberry The generated proposal will not add prefixes to their title if they are at either the `community` or `default` stability level.

Is that ok or do you want to display one or the other?

Comment (bstansberry) [reply _includes/index-categories.html:32]: @jmesnil We should display community.

**Review by bstansberry — CHANGES_REQUESTED** [review]
Hi @jmesnil -- I have a couple requests.

For context around these requests please have a look at https://docs.google.com/document/d/1e2hCAb--ApK32fPbUTWQSn6fH9GqmqZuaUPYwIp01JQ/edit. That's a draft of an email I plan to send to wildfly-dev once we merge this.

File: _layouts/proposal.html [hunk 11]
```diff
@@ -14,6 +19,9 @@ <h1 class="title">{{page.title}}</h1>
     {% endfor %}
     </div>
     {% endunless %}
+    <div style="padding-bottom: 2em;">Tracked by&nbsp;
+      <a href="{{ page.issue }}">{{ page.issue }}</a>
```
Comment (bstansberry) [inline _layouts/proposal.html:23]: The first '{{ page.issue }}' has a special char (NBSP) before the 'page' which messes with the jekyll site rendering. So please replace with a regular space.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: design-doc-template.adoc [hunk 12]
```diff
@@ -4,119 +4,92 @@ categories:
 # - core
 # - management
 # if missing, add it to _data/widfly-categories and use its id
+#
+# Specify the stability level of the feature.
+# Values can be one of: experimental preview community default
+stability-level: 
+# Specify the main issue of the feature
+# Most of the time, it will be a issue in https://issues.redhat.com/browse/WFLY
+# or https://issues.redhat.com/browse/WFCORE
```
[the remark this answers is not in the document]
Comment (jmesnil) [reply design-doc-template.adoc]: done, thanks

File: design-doc-template.adoc [hunk 13]
```diff
@@ -4,119 +4,92 @@ categories:
...
-
-== Issue Metadata
-
-=== Issue
-
-* https://issues.redhat.com/browse/WFCORE[WFCORE-XXXX]
-
-=== Related Issues
-
-* https://issues.redhat.com/browse/WFLY[WFLY-XXXX]
```
[the remark this answers is not in the document]
Comment (jmesnil) [reply design-doc-template.adoc:26]: I brought back the `Related Issues` section

File: _includes/index-categories.html (hunk 10 above)
Comment (jmesnil) [reply _includes/index-categories.html]: done (both in the proposal pages and the index page)

**Comment by jmesnil** [comment]
@bstansberry I've updated the PR and made some suggestions to the mail you will send. thanks!

**Comment by jmesnil** [comment]
I've created the labels that will be added to the PR based on their stability-level attributes:
* `stability-level/experimental`
* `stability-level/preview`
* `stability-level/community`
* `stability-level/default`

**Comment by jmesnil** [comment]
@bstansberry Could we also merge https://github.com/wildfly/wildfly-proposals/pull/584? It would unclutter the PR queue.
That would close 56 of 85 PR that have not been updated for the last 4 months (https://github.com/wildfly/wildfly-proposals/pulls?q=is%3Apr+updated%3A%3C2024-06-06+is%3Aopen)

**Comment by bstansberry** [comment]
Thanks @jmesnil

**Pull request merged**